pygbm/splitting.py

```diff
@@ -61,66 +64,96 @@ def __init__(self, n_features, binned_features, n_bins,
         else:
             self.constant_hessian_value = float32(1.)  # won't be used anyway
 
+        # The partition array maps each sample index into the leaves of the
+        # tree (a leaf in this context is a node that isn't splitted yet, not
+        # necessarily a 'finalized' leaf). Initially, the root contains all
+        # the indices, e.g.:
+        # partition = [abcdefghijkl]
+        # After a call to split_indices, it may look e.g. like this:
+        # partition = [cef|abdghijkl]
+        # we have 2 leaves, the left one is at position 0 and the second one at
+        # position 3. The order of the samples is irrelevant.
+        self.partition = np.arange(0, binned_features.shape[0], 1, np.uint32)
+        # tmp arrays are used in split_indices to support parallel splitting.
+        self.tmp_left = np.empty_like(self.partition)
+        self.tmp_right = np.empty_like(self.partition)
```

Comment:
Maybe it's possible to find a more descriptive name for those attributes: `left_indices_buffer`  and `right_indices_buffer`.